natlas-server/app/cyclicprng.py

```diff
@@ -0,0 +1,89 @@
+from threading import Lock
+import random
+import sympy
```

Comment:
`sympy` is a new dependency that isn't in the python standard libraries, so this will require an updated requirements.txt

Reply:
0xdade did you say you were going to make this change locally? I've refrained from touching/checking-in requirements.txt.

Reply:
You should update your requirements.txt and include it in this PR if you can - that way the merge will be an atomic unit to implement the functionality. (I haven't always been good about this in the past, committing things that I forgot dependencies for)